accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -94,20 +94,21 @@ library section below.
 
 ### Static errors
 
-We say that a type `T` is **nullable** if `Null <: T`.  This is equivalent to
-the syntactic criterion that `T` is any of:
+We say that a type `T` is **nullable** if `Null <: T` and not `T <: Object`.
+This is equivalent to the syntactic criterion that `T` is any of:
   - `Null`
   - `S?` for some `S`
   - `S*` for some `S` where `S` is nullable
   - `FutureOr<S>` for some `S` where `S` is nullable
   - `dynamic`
   - `void`
 
-We say that a type `T` is **non-nullable** if `T <: Object`.  This is equivalent
-to the syntactic criterion that `T` is any of:
-  - `Object`, `int`, `bool`, `Never`, `Function`
-  - Any function type
-  - Any class type or generic class type
+We say that a type `T` is **non-nullable** if `T <: Object`.
+This is equivalent to the syntactic criterion that `T` is any of:
+  - `Never`
+  - Any function type (including `Function`)
+  - Any interface type except `dynamic`, `void`, and `Null`.
```

Comment:
Nitpickery: I don't consider `Function` a function type, nor `dynamic` or `void` as interface types. :)

Reply:
Is Null an interface type?

Reply:
So far it has been. It behaves just as any other (sealed) interface type, so there hasn't been any need to special-case it.

Reply:
Well, it is special cased in the type system - you can't get the right subtyping otherwise.  That's why I get confused... :)